File: Software-Praktikum/Frontend/lernapp/src/API/GroupBO.js
import BusinessObject from "./BusinessObject.js"

/**
 *  @author [Khadidja Kebaili](https://github.com/Khadidja-Kebaili)
 *  @coauthor: [Lena Tolamtschow] (https://github.com/LenaTolmatschow)
 */

export default class GroupBO extends BusinessObject{

    /**
     *
     * @param {string} groupname - Id einer ProfileBO
     * @param {int} admin - Id des Admins
     * @param {string} description - Gruppenbeschreibung
     * @param {int} chatid - ID des zugegörigen Chatrooms.
     */

    constructor(groupname, admin, description, chatid) {
        super();
        this.groupname = groupname
        this.admin = admin
        this.description = description
        this.chatid = chatid

    }

    /**
     * Setzt Gruppenname
     */

    setGroupname(groupname) {
        this.groupname = groupname;
    }

    /**
     * Gibt Gruppennamen zurück
     */

    getGroupname() {
        return this.groupname;
    }

    /**
     * Setzt Gruppenadmin
     * @param admin
     */
    setAdmin (admin) {
        this.admin = admin;
    }

    /**
     * Gibt Gruppenadmin zurück
     * @returns {Number}
     */
    getAdmin() {
        return this.admin;
    }

    /**
     * Setzt Gruppenbeschreibung
     * @param description
     */
    setDescription (description) {
        this.description = description;
    }

    /**
     * gibt Gruppenbeschreibung zurück
     * @returns {string}
     */
    getDescription() {
        return this.description;
    }

    /**
     * Setzt die zugehörige Chatid
     * @param chatid
     */
    setChatid (chatid) {
        this.chatid = chatid;
    }

    /**
     * Gibt zugehörige Chatid zurück
     * @returns {Number}
     */
    getChatid() {
        return this.chatid;
    }

    /**
     * Nimmt JSON-Objekte und wandelt sie in einen Array um mit GroupBOs.
     * @param group
     * @returns {[GroupBO]}
     */
    static fromJSON(group) {
        let result = [];

        if (Array.isArray(group)) {
            group.forEach((c) => {
                Object.setPrototypeOf(c, GroupBO.prototype);
                result.push(c);
            })
        } else {
            // Es handelt sich offenbar um ein singuläres Objekt
            let c = group;
            Object.setPrototypeOf(c, GroupBO.prototype);
            result.push(c);
        }

        return result;
    }
}

File: Software-Praktikum/src/server/bo/GroupBO.py
from src.server.bo.Businessobject import Businessobject

class Group(Businessobject):
    def __init__(self):
        super().__init__()

        self.description = ""
        self.groupname = "",
        self.admin = "",
        self.chatid = "",

    def set_description(self, value):
        self.description = value

    def get_description(self):
        return self.description

    def set_groupname(self, value):
        self.groupname = value

    def get_groupname(self):
        return self.groupname

    def set_admin(self, value):
        self.admin = value

    def get_admin(self):
        return self.admin

    def set_chatid(self, value):
        self.chatid = value

    def get_chatid(self):
        return self.chatid


    def __str__(self):
        """Erzeugen einer einfachen textuellen Darstellung der jeweiligen Instanz."""
        return "Info:{}, {}, {}, {}".format(self.get_description(), self.get_groupname(), self.get_admin(),
                                            self.get_chatid())

    @staticmethod
    def from_dict(dictionary=dict()):
        """Umwandeln eines Python dict() in einen User()."""
        obj = Group()
        obj.set_id(dictionary["id"])
        obj.set_description(dictionary["description"])
        obj.set_groupname(dictionary["groupname"])
        obj.set_admin(dictionary["admin"])
        obj.set_chatid(dictionary["chatid"])
        return obj
